pipeline/leech/main.go

```diff
@@ -138,6 +157,44 @@ func realMain(ctx context.Context) error {
 	return nil
 }
 
+// Validates inputs which are used in BigQuery string interpolation to help avoid SQL-Injection.
+func validateInputs(eventsProjectID *string, eventsTable *string, leechProjectID *string, leechTable *string) error {
+	var merr error
+	if err := bigquery.ValidateGCPProjectID(*eventsProjectID); err != nil {
+		merr = errors.Join(merr, fmt.Errorf("invalid events-project-id arg: %w", err))
+	}
+	eventsTableDataset, eventsTableName, err := splitDatasetTableName(*eventsTable)
```

Comment:
This could perhaps be simplified by factoring out a function. The presence of "else" and the presence of copy-pasted blocks of code with variable names changed are both hints that factoring out a function could be beneficial. I'm thinking something like:

```
func splitAndValidate(dottedTableName string) (dataset string, table string, _ error) {
  dataset, table, err := splitDatasetTableName(*eventsTable)
  if err != nil {
    return err
  }
  return errors.Join(
    bigquery.ValidateDatasetID(eventsTableDataset),
    bigquery.ValidateTableName(eventsTableName))
}
```

And you might even be able to inline `splitDatasetTableName` into this function for more simplicity.